staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -986,6 +1018,52 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 	return allErrs
 }
 
+func extractMaxElements(schema *apiextensions.JSONSchemaProps) *uint64 {
+	switch schema.Type {
+	case "object":
+		if schema.AdditionalProperties != nil && schema.MaxProperties != nil {
+			if schema.MaxProperties != nil {
+				maxProps := uint64(*schema.MaxProperties)
+				return &maxProps
+			}
+			return nil
+		}
+		// return 1, and not nil, because even if we're
+		// not looking at a map here, we can assume the object still preserves
+		// the cardinality of its parents
+		cardOne := uint64(1)
+		return &cardOne
+	case "array":
+		if schema.MaxItems != nil {
+			maxItems := uint64(*schema.MaxItems)
+			return &maxItems
+		}
+		return nil
+	default:
+		return nil
+	}
+}
+
+// multWithOverflowGuard is intended to serve as a convenience function to handle the potential for
```

Comment:
Consider saying crisply what the function does. E.g. "multWithOverflowGuard returns the product of baseCost and cardinality unless the product exceeds math.MaxUInt, in which case math.MaxUInt is returned.

Reply:
Gotcha, I've reworded the documentation